stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/WriteCharacters.java

```diff
@@ -61,29 +71,75 @@
 
     /**
-     * Writes characters to a given file.
+     * Will be the I/O event handler.
+     */
+    private EventManager eventManager = EventManager.getInstance();
+
+    private static final Logger log = LoggerFactory.getLogger(WriteCharacters.class);
+
+/*
+    private static EventResult readCharactersResponse(EventResult<Integer, EventContext> result) {
+        BStruct errorStruct;
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        Integer numberOfBytes = result.getResponse();
+        return result;
+    }
+*/
+
+    /**
+     * Writes characters asynchronously.
      *
+     * @param characterChannel channel the characters should be written.
+     * @param text             the content which should be written.
+     * @param offset           the index the characters should be written.
+     * @param context          context of the event.
+     * @return the number of characters which was written.
+     * @throws ExecutionException   errors which occur during execution.
+     * @throws InterruptedException during interrupt exception.
+     */
+    private int writeCharacters(CharacterChannel characterChannel, String text, int offset, EventContext context)
+            throws ExecutionException, InterruptedException {
+        WriteCharactersEvent event = new WriteCharactersEvent(characterChannel, text, offset, context);
+        CompletableFuture<EventResult> future = eventManager.publish(event);
+        EventResult eventResult = future.get();
+        Throwable error = ((EventContext) eventResult.getContext()).getError();
+        if (null != error) {
+            throw new ExecutionException(error);
+        }
+        return (int) eventResult.getResponse();
+    }
+
+    /**
+     * Writes characters to a given file.
+     * <p>
      * {@inheritDoc}
      */
     @Override
     public BValue[] execute(Context context) {
         BStruct channel;
         String content;
         long startOffset;
-        int numberOfCharactersWritten;
+        int numberOfCharactersWritten = 0;
+        BStruct errorStruct = null;
         try {
             channel = (BStruct) getRefArgument(context, CHAR_CHANNEL_INDEX);
             content = getStringArgument(context, CONTENT_INDEX);
             startOffset = getIntArgument(context, START_OFFSET_INDEX);
-
             CharacterChannel characterChannel = (CharacterChannel) channel.getNativeData(IOConstants
                     .CHARACTER_CHANNEL_NAME);
-
-            numberOfCharactersWritten = characterChannel.write(content, (int) startOffset);
+            EventContext eventContext = new EventContext(context);
+//            IOUtils.write(characterChannel,content,startOffset,WriteCharacters::readCharactersResponse);
```

Comment:
Remove commented line

Reply:
This needs to be uncommented upon the availability of the async functions